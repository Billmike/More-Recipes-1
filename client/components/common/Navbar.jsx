import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import Category from './Category';
import IndexUserNav from './IndexUserNav';
import AuthUserNav from './AuthUserNav';
import Signup from './Signup';
import Signin from './Signin';

const propTypes = {
  handleSubmitSignup: PropTypes.func.isRequired,
  handleSubmitSignin: PropTypes.func.isRequired,
  handleLogoutUser: PropTypes.func.isRequired,
  user: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired,
    userAuthentication: PropTypes.shape({
      fullName: PropTypes.string
    }),
    userProfile: PropTypes.shape({
      fullName: PropTypes.string,
    }),
  }).isRequired,
};


/**
 * Class representing navbar component
 *
 * @class Navbar
 *
 * @extends {React.Component}
 */
class Navbar extends React.Component {
  /**
   * Component did update life cycle method
   *
   * @returns {any} Initialize materialize component
   *
   * @memberof Navbar
   */
  componentDidUpdate() {
    // Initialize materialize dropdown class
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
  }

  /**
   * Render method
   *
   * @returns{object} React element
   *
   * @memberof Navbar
   */
  render() {
    const {
      user: { isAuthenticated, userAuthentication }
    } = this.props;
    const { fullName } = userAuthentication;

    return (
      <div className="nav-wrapper">
        <Link
          to="/"
          id="logo"
          className="brand-logo hide-on-small-only"
        >
          More-Recipes
        </Link>
        <a
          data-activates="slide_out"
          className="button-collapse"
        >
          <i className="material-icons">menu</i>
        </a>
        <ul className="right hide-on-med-and-down">
          <li><Link to="/">Home</Link></li>
          <li><Link to="/catalog">Catalog</Link></li>
          <li>
            <a
              className="dropdown-button dropdown-category"
              data-activates="category"
            >
              Category
            </a>
          </li>
          <Category {...this.props} />
          <li>
            <a
              className="dropdown-button dropdown-user"
              data-activates="user-control"
            >
              {!isAuthenticated
                ? 'Welcome Guest'
                : fullName || localStorage.getItem('fullName')
              }
              <i className="material-icons large left">account_circle</i>
            </a>
            {!isAuthenticated &&
              <IndexUserNav {...this.props} />
            }
            {isAuthenticated &&
              <AuthUserNav {...this.props} />}
          </li>
        </ul>
        <MuiThemeProvider>
          <Signup
            {...this.props}
          />
        </MuiThemeProvider>
        <MuiThemeProvider>
          <Signin
            {...this.props}
          />
        </MuiThemeProvider>
      </div>
    );
  }
}

Navbar.propTypes = propTypes;

export default Navbar;
