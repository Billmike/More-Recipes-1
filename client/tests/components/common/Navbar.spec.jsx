import React from 'react';

import Navbar from '../../../components/common/Navbar';

describe('Navbar component', () => {
  it('should render component', () => {
    const props = {
      handleSubmitSignup: jest.fn(),
      handleSubmitSignin: jest.fn(),
      handleLogoutUser: jest.fn(),
      user: {
        isAuthenticated: false,
        userAuthentication: {},
        userProfile: {},
      },
    };
    const wrapper = shallow(<Navbar {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate component did update', () => {
    const props = {
      handleSubmitSignup: jest.fn(),
      handleSubmitSignin: jest.fn(),
      handleLogoutUser: jest.fn(),
      user: {
        isAuthenticated: false,
        userAuthentication: {},
        userProfile: {},
      },
    };
    const nextProps = {
      user: {
        isAuthenticated: true,
        userAuthentication: {
          fullName: 'John Mark',
        },
        userProfile: {
          firstName: 'John',
          lastName: 'Mark',
        },
      },
    };
    const spy = jest.spyOn(Navbar.prototype, 'componentDidUpdate');
    const wrapper = shallow(<Navbar {...props} />);
    expect(wrapper).toBeDefined();
    wrapper.setProps(nextProps);
    expect(spy).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
