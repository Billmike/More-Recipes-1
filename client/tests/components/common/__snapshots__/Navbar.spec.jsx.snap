// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render component 1`] = `
<div
  className="nav-wrapper"
>
  <Link
    className="brand-logo hide-on-small-only"
    id="logo"
    replace={false}
    to="/"
  >
    More-Recipes
  </Link>
  <a
    className="button-collapse"
    data-activates="slide_out"
  >
    <i
      className="material-icons"
    >
      menu
    </i>
  </a>
  <ul
    className="right hide-on-med-and-down"
  >
    <li>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <a
        className="dropdown-button dropdown-category"
        data-activates="category"
      >
        Category
      </a>
    </li>
    <Category
      handleLogoutUser={[MockFunction]}
      handleSubmitSignin={[MockFunction]}
      handleSubmitSignup={[MockFunction]}
      user={
        Object {
          "isAuthenticated": false,
          "userAuthentication": Object {},
          "userProfile": Object {},
        }
      }
    />
    <li>
      <a
        className="dropdown-button dropdown-user"
        data-activates="user-control"
      >
        Welcome Guest
        <i
          className="material-icons large left"
        >
          account_circle
        </i>
      </a>
      <IndexUserNav
        handleLogoutUser={[MockFunction]}
        handleSubmitSignin={[MockFunction]}
        handleSubmitSignup={[MockFunction]}
        user={
          Object {
            "isAuthenticated": false,
            "userAuthentication": Object {},
            "userProfile": Object {},
          }
        }
      />
    </li>
  </ul>
  <MuiThemeProvider>
    <ReduxForm
      handleLogoutUser={[MockFunction]}
      handleSubmitSignin={[MockFunction]}
      handleSubmitSignup={[MockFunction]}
      user={
        Object {
          "isAuthenticated": false,
          "userAuthentication": Object {},
          "userProfile": Object {},
        }
      }
    />
  </MuiThemeProvider>
  <MuiThemeProvider>
    <ReduxForm
      handleLogoutUser={[MockFunction]}
      handleSubmitSignin={[MockFunction]}
      handleSubmitSignup={[MockFunction]}
      user={
        Object {
          "isAuthenticated": false,
          "userAuthentication": Object {},
          "userProfile": Object {},
        }
      }
    />
  </MuiThemeProvider>
</div>
`;

exports[`Navbar component should simulate component did update 1`] = `
<div
  className="nav-wrapper"
>
  <Link
    className="brand-logo hide-on-small-only"
    id="logo"
    replace={false}
    to="/"
  >
    More-Recipes
  </Link>
  <a
    className="button-collapse"
    data-activates="slide_out"
  >
    <i
      className="material-icons"
    >
      menu
    </i>
  </a>
  <ul
    className="right hide-on-med-and-down"
  >
    <li>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <a
        className="dropdown-button dropdown-category"
        data-activates="category"
      >
        Category
      </a>
    </li>
    <Category
      handleLogoutUser={[MockFunction]}
      handleSubmitSignin={[MockFunction]}
      handleSubmitSignup={[MockFunction]}
      user={
        Object {
          "isAuthenticated": true,
          "userAuthentication": Object {
            "fullName": "John Mark",
          },
          "userProfile": Object {
            "firstName": "John",
            "lastName": "Mark",
          },
        }
      }
    />
    <li>
      <a
        className="dropdown-button dropdown-user"
        data-activates="user-control"
      >
        John Mark
        <i
          className="material-icons large left"
        >
          account_circle
        </i>
      </a>
      <AuthUserNav
        handleLogoutUser={[MockFunction]}
        handleSubmitSignin={[MockFunction]}
        handleSubmitSignup={[MockFunction]}
        user={
          Object {
            "isAuthenticated": true,
            "userAuthentication": Object {
              "fullName": "John Mark",
            },
            "userProfile": Object {
              "firstName": "John",
              "lastName": "Mark",
            },
          }
        }
      />
    </li>
  </ul>
  <MuiThemeProvider>
    <ReduxForm
      handleLogoutUser={[MockFunction]}
      handleSubmitSignin={[MockFunction]}
      handleSubmitSignup={[MockFunction]}
      user={
        Object {
          "isAuthenticated": true,
          "userAuthentication": Object {
            "fullName": "John Mark",
          },
          "userProfile": Object {
            "firstName": "John",
            "lastName": "Mark",
          },
        }
      }
    />
  </MuiThemeProvider>
  <MuiThemeProvider>
    <ReduxForm
      handleLogoutUser={[MockFunction]}
      handleSubmitSignin={[MockFunction]}
      handleSubmitSignup={[MockFunction]}
      user={
        Object {
          "isAuthenticated": true,
          "userAuthentication": Object {
            "fullName": "John Mark",
          },
          "userProfile": Object {
            "firstName": "John",
            "lastName": "Mark",
          },
        }
      }
    />
  </MuiThemeProvider>
</div>
`;
